lib/Core/Executor.cpp

```diff
@@ -3984,20 +3986,19 @@ static std::set<std::string> okExternals(okExternalsList,
                                          okExternalsList + 
                                          (sizeof(okExternalsList)/sizeof(okExternalsList[0])));
 
-void Executor::callExternalFunction(ExecutionState &state,
-                                    KInstruction *target,
+void Executor::callExternalFunction(ExecutionState &state, KInstruction *target,
                                     KCallable *callable,
-                                    std::vector< ref<Expr> > &arguments) {
+                                    std::vector<ref<Expr>> &arguments) {
   // check if specialFunctionHandler wants it
-  if (const auto *func = dyn_cast<KFunction>(callable)) {
-    if (specialFunctionHandler->handle(state, func->function, target, arguments))
-      return;
-  }
+  const auto *func = dyn_cast<KFunction>(callable);
```

Comment:
Please keep `func` scoped like in the original implementation. Using the "initialiser if" will help with your use case, i.e.:
```suggestion
  if (const auto *func = dyn_cast<KFunction>(callable); func &&
      specialFunctionHandler->handle(state, func->function, target, arguments))
    return;
```